### ISSUE TITLE
Add pyoptsparse version in the sqlitedict for checkpointing

### ISSUE THREAD

**Issue by johnjasa** [mdolab/pyoptsparse#42]
## Type of issue
What types of issue is it?
- New feature (non-breaking change which adds functionality)

## Description
Add the git commit hash for the pyoptsparse version that was used to run the case. This makes it easier to examine or debug cases from the past or when sending them to colleagues.

See this stackoverflow answer for a quick fix:
https://stackoverflow.com/questions/14989858/get-the-current-git-hash-in-a-python-script

```
import subprocess

def get_git_revision_hash():
    return subprocess.check_output(['git', 'rev-parse', 'HEAD'])

def get_git_revision_short_hash():
    return subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD'])
```

### Current behavior
The git commit hash is not recorded

### Expected behavior
We want to save the recorded git hash

**Comment by nwu63** [comment]
@johnjasa the current plan is to just store the version of pyOptSparse itself, since we started version tagging. I have been operating under the assumption that anything that modifies the behavior of the code, rather than a pure "new feature" PR would increment the version. Would this be sufficient instead of the git commit hash?

**Comment by johnjasa** [comment]
I think storing both the version and the git commit hash would be best, since that would rule out any potential changes in effects due to even incremental changes. In a perfect world, we would version it every time there is any change that could potentially alter behavior, but I'm not certain if I'd rely on that in practice. Do you think it'd be an undue burden to save both pieces of information?

**Comment by nwu63** [comment]
Yeah fair, I'll try out the script you wrote above and give it a shot.

**Comment by eirikurj** [comment]
Addressed in #77, closing.
